garrysmod/lua/includes/extensions/util.lua

```diff
@@ -365,3 +365,53 @@ function util.RemovePData( steamid, name )
 	sql.Query( "DELETE FROM playerpdata WHERE infoid = " .. SQLStr( name ) )
 
 end
+
+--[[---------------------------------------------------------
+	Name: StringToBinary( str )
+	Desc: Converts a string to a binary string
+-----------------------------------------------------------]]
+
+function util.StringToBinary( str )
+
+	local bin = ""
+	str = str:reverse()
+
+	for i = 1, str:len() do
+		local num = str[ i ]:byte()
+		for j = 7, 0, -1 do -- 8 bits in each char
+			bin = bin .. math.floor( num / 2 ^ j )
+			num = num % 2 ^ j
+		end
+
+		bin = bin .. " "

```

Comment:
I believe using a table to build strings is faster than repeated concatenation(?)

```lua
local str = {}
str[#str + 1] = "..."
...
str = table.concat(str)
```